src/whenever/__init__.py

```diff
@@ -390,10 +389,16 @@ def from_canonical_format(cls, s: str, /) -> Date:
         >>> Date.from_canonical_format("2021-01-02")
         Date(2021-01-02)
         """
-        if s[5] == "W":
-            # prevent isoformat from parsing week dates
-            raise InvalidFormat()
-        return cls.from_py_date(_date.fromisoformat(s))
+        try:
+            if s[5] == "W":
+                # prevent isoformat from parsing week dates
+                raise ValueError("Week dates are not supported")
```

Comment:
Let's just stick with the normal exception message—as far as the user should be concerned, all non `YYYY-MM-DD` are invalid. 

The only reason for this separate check is that `fromisoformat` will parse week dates, and we don't want to.

Reply:
Done.